bcda/models/models.go

```diff
@@ -40,14 +45,93 @@ func InitializeGormModels() *gorm.DB {
 type Job struct {
 	gorm.Model
 	ACO        ACO       `gorm:"foreignkey:ACOID;association_foreignkey:UUID"` // aco
-	ACOID      uuid.UUID `gorm:"primary_key; type:char(36)" json:"aco_id"`
+	ACOID      uuid.UUID `gorm:"type:char(36)" json:"aco_id"`
 	User       User      `gorm:"foreignkey:UserID;association_foreignkey:UUID"` // user
 	UserID     uuid.UUID `gorm:"type:char(36)"`
 	RequestURL string    `json:"request_url"` // request_url
 	Status     string    `json:"status"`      // status
+	JobCount   int
 	JobKeys    []JobKey
 }
 
+func (job *Job) CheckCompleted() (bool, error) {
+
+	// Trivial case, no need to keep going
+	if job.Status == "Completed" {
+		return true, nil
+	}
+	db := database.GetGORMDbConnection()
+	defer database.Close(db)
+
+	var completedJobs int64
+	db.Model(&JobKey{}).Where("job_id = ?", job.ID).Count(&completedJobs)
+
+	if int(completedJobs) >= job.JobCount {
```

Comment:
Is it possible for there to be more completed jobs than the count?

Reply:
I don't think so, but I get nervous about 0 vs 1 indexed things and I didn't want to get into a weird permanent pending status if they did have more.